src/datasets/graphs/movielens.jl

```diff
@@ -0,0 +1,155 @@
+function __init__movielens()
+    DEPNAME = "MovieLens"
+    DOCS = "https://grouplens.org/datasets/movielens/"
+
+    register(ManualDataDep(
+        DEPNAME,
+        """
+        Datahub: $DEPNAME.
+        Website: $DOCS
+        """,
+    ))
+end
+
+"""
+    MovieLens(name; dir=nothing)
+"""
+struct MovieLens
+  name::String
+  metadata::Dict{String, Any}
+  graphs::Vector{HeteroGraph}
+end
+
+function MovieLens(name::String; dir=nothing)
+    name = lowercase(name)
+    create_default_dir("MovieLens")
+    dir = movielens_datadir(name, dir)
+    # format varied in older generations of MovieLens datasets
+    if name == "100k"
+      user_data, movie_data, rating_data = read_100k_data(dir)
+      g = generate_movielens_graph(user_data, movie_data, rating_data)
+      metadata = get_movielens_metadata([user_data, movie_data, rating_data])
+      return MovieLens(name, metadata, [g])
+    else
+      error("Functionality for ml-$name has not been implemented yet")
+    end
+end
+
+function read_100k_data(dir::String)
+  user_data = read_100k_user_data(dir)
+  movie_data = read_100k_movie_data(dir)
+  rating_data = read_100k_rating_data(dir)
+  return (user_data, movie_data, rating_data)
+end
+
+function read_100k_user_data(dir::String)::Dict
+  user_data_file = "u.user"
+  user_data = Dict()
+  user_df = read_csv_asdf(joinpath(dir, user_data_file), header=false)
+
+  user_ids = user_df[!, 1]
+  @assert minimum(user_ids) == 1
+  user_data["num_users"] = maximum(user_ids)
+  user_data["age"] = user_df[!, 2]
+  user_data["gender"] = user_df[!, 3] .== "M" # I hope I don't get cancelled for binarizing this field
+  user_data["occupation"] = user_df[!, 4]
+  user_data["zipcode"] = user_df[!, 5]
+  return user_data
```

Comment:
The indentation is not uniform in this file. It should be 4 blanks everywhere

Reply:
Yeah, vim messed it up somewhere, I will fix these things in a cleanup commit later.